morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -0,0 +1,397 @@
+
+import Cardano.BM.Data.Tracer (TracingVerbosity (..))
+import Cardano.Crypto.DSIGN.Class (Signable)
+import Cardano.Crypto.Hash.Class (Hash, HashAlgorithm, hashToTextAsHex)
+import Cardano.Prelude hiding (list)
+import qualified Data.Text as Text
+import Morpho.Ledger.Block (Header (..), MorphoBlock (..), MorphoBlockTx (morphoBlockGenTx), MorphoBody (..), MorphoStdHeader (..))
+import Morpho.Ledger.PowTypes (Vote)
+import Morpho.Ledger.Tx (Tx (Tx))
+import Morpho.Ledger.Update (GenTx, MorphoStateDefaultConstraints)
+import Morpho.Node.RunNode ()
+import Network.Mux (MuxTrace, WithMuxBearer)
+import qualified Network.Socket as Socket (SockAddr)
+import Ouroboros.Consensus.Block (ChainHash, HeaderHash, Point (..), StandardHash, blockHash, headerPoint, pointSlot, realPointSlot)
+import Ouroboros.Consensus.Block.Abstract (BlockNo (..), SlotNo (..))
+import Ouroboros.Consensus.BlockchainTime (SystemStart (..), TraceBlockchainTimeEvent (..))
+import Ouroboros.Consensus.Ledger.SupportsMempool (ApplyTxErr, GenTxId, txId)
+import Ouroboros.Consensus.Ledger.SupportsProtocol (LedgerSupportsProtocol)
+import Ouroboros.Consensus.Mempool (TraceEventMempool)
+import Ouroboros.Consensus.MiniProtocol.BlockFetch.Server (TraceBlockFetchServerEvent)
+import Ouroboros.Consensus.MiniProtocol.ChainSync.Client (TraceChainSyncClientEvent (..))
+import Ouroboros.Consensus.MiniProtocol.ChainSync.Server (TraceChainSyncServerEvent)
+import Ouroboros.Consensus.MiniProtocol.LocalTxSubmission.Server (TraceLocalTxSubmissionServerEvent)
+import Ouroboros.Consensus.Node.Tracers (TraceLabelCreds (TraceLabelCreds))
+import Ouroboros.Consensus.NodeKernel (TraceForgeEvent (..))
+import Ouroboros.Consensus.Protocol.BFT (BftCrypto (BftDSIGN))
+import qualified Ouroboros.Consensus.Storage.ChainDB as ChainDB
+import qualified Ouroboros.Consensus.Storage.LedgerDB.OnDisk as LedgerDB
+import Ouroboros.Consensus.Util.Condense (Condense, condense)
+import qualified Ouroboros.Network.AnchoredFragment as AF
+import Ouroboros.Network.BlockFetch.ClientState
+  ( TraceLabelPeer (..),
+  )
+import Ouroboros.Network.Diffusion (DiffusionInitializationTracer)
+import qualified Ouroboros.Network.NodeToClient as NtC
+import Ouroboros.Network.NodeToNode
+  ( DnsTrace (..),
+    ErrorPolicyTrace (..),
+    SubscriptionTrace (..),
+    WithAddr (..),
+    WithDomainName (..),
+    WithIPList (..),
+  )
+import qualified Ouroboros.Network.NodeToNode as NtN
+import Ouroboros.Network.PeerSelection.LedgerPeers (TraceLedgerPeers)
+import Ouroboros.Network.TxSubmission.Inbound (TraceTxSubmissionInbound)
+import Ouroboros.Network.TxSubmission.Outbound (TraceTxSubmissionOutbound)
+import Prettyprinter (Doc, Pretty (..), list, viaShow, (<+>))
+import Prelude (String, id)
+
+class MPretty a where
+  mpretty :: a -> Doc ann
+
+instance (HashAlgorithm h, BftCrypto c) => MPretty (MorphoBlock h c) where
+  mpretty MorphoBlock {..} =
+    "("
+      <> mpretty morphoPrev
+      <> "->"
```

Comment:
Currently a block is printed as
```
(BlockHash "73417a76a6e740a2"->ab7f92b6,315584,[  ])
```

Needs some more labels like [here](https://github.com/input-output-hk/ECIP-Checkpointing/compare/d3983944c3587990a738390ce4e1f8f0cd4f4129...e50939d7831dc42e6cfe736e208845ba64b948f4#diff-7d5bb714443018594f3f29a5648fc551d8f9f516a561da5dada77270f127dad8R283)

Same for `Header` below.

We could also just use `viaShow` for these for now.

Reply:
I copied/pasted the old implementation. We can change this in a new commit by integrating what was done by your commit. Is that okay?

Reply:
Ohh didn't even realize the previous one was this bad, yeah sounds good then